Automattic/wp-calypso
# ETK: Remove post published modal

Related to https://github.com/Automattic/wp-calypso/issues/72803

## Estimated Time to Test / Review:
- Test -> Short / Med
- Review -> Short

## Proposed Changes

There are multiple modals that want to respond to the post editor's publish / save / update button. To address this, we're consolidating all modal logic into the Jetpack repo. This has the added benefit of ensuring that the first post published modal will _always_ appear, because, while the user _can_ disable the editing toolkit plugin, they _cannot_ disable the jetpack features we're implementing

* Remove first post published modal and all references to it from Calypso

## Testing Instructions



* Check out this branch
* [Create a new site through general onboarding](wordpress.com/start)
* Select `Write & Publish` as the site's goal
* Select any category
* Use any site title and tagline
* Before moving forward, sandbox your site
* Navigate to `apps/editing-toolkit` in a terminal
*  `yarn dev --sync`
* Continue through onboarding and Click on the `Start writing` button
* Edit the post 
* Click on the publish button
* Verify that no first post published modal appears. Only the launchpad modal will render for the time being.

## GIFs
### Before

![2023-02-14 11 56 09](https://user-images.githubusercontent.com/5414230/218847442-823176e1-e4ba-4beb-ba69-7cfadd9e5024.gif)

### After

![2023-02-14 11 53 48](https://user-images.githubusercontent.com/5414230/218847037-593ae4dd-665c-4128-a62b-124b712b60ca.gif)


## Pre-merge Checklist
